zigzag/datatypes.py

```diff
@@ -102,9 +102,15 @@ class Constants:
     MEM_OP_2 = MemoryOperand(AcceleratorValidator.MEM_OP_2_STR)
 
 
+class UnknownUnrollFactor:
```

Comment:
Instead of creating an empty class and converting the `*` values to an instance of this class, you could also define `UnknownUnrollFactor: TypeAlias = Literal["*"]` and omit the conversion altogether